File: web/src/utils/arabicFonts.js
/**
 * CONVERSION: Mirrors lib/arabic_fonts.dart — Amiri as Google Fonts fallback;
 * KFGQPCUthmanTahaNaskh if you register @font-face in index.css (kUseBundledKfgqpcFont).
 */
export const kKfgqpcUthmanTahaNaskhFamily = 'KFGQPCUthmanTahaNaskh';

/** Set false to use only Amiri (same toggle intent as Dart). */
export const kUseBundledKfgqpcFont = true;

/**
 * @param {object} opts
 * @param {number} [opts.fontSize]
 * @param {number|string} [opts.fontWeight]
 * @param {string} [opts.color]
 * @param {number} [opts.lineHeight] — maps to Flutter TextStyle.height
 */
export function arabicQuranTextStyle({
  fontSize,
  fontWeight,
  color,
  lineHeight,
} = {}) {
  const fontFamily = kUseBundledKfgqpcFont
    ? `'${kKfgqpcUthmanTahaNaskhFamily}', 'Amiri', serif`
    : `'Amiri', serif`;

  return {
    fontFamily,
    fontSize: fontSize != null ? `${fontSize}px` : undefined,
    fontWeight: fontWeight ?? undefined,
    color: color ?? undefined,
    lineHeight: lineHeight != null ? String(lineHeight) : undefined,
  };
}

File: lib/arabic_fonts.dart
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';

/// Quran-style Arabic font family name.
///
/// When you add the KFGQPC font file to your project and register it in
/// `pubspec.yaml` under `flutter: fonts:`, keep the `family:` value exactly the
/// same as this constant.
const String kKfgqpcUthmanTahaNaskhFamily = 'KFGQPCUthmanTahaNaskh';

/// Toggle to use the bundled KFGQPC font (once you've added & registered it).
///
/// Set this to true to use the local KFGQPC font family registered in
/// `pubspec.yaml`. When false, we use a Google Fonts Arabic fallback (Amiri).
const bool kUseBundledKfgqpcFont = true;

TextStyle arabicQuranTextStyle({
  double? fontSize,
  FontWeight? fontWeight,
  Color? color,
  double? height,
}) {
  final fallback = GoogleFonts.amiri(
    fontSize: fontSize,
    fontWeight: fontWeight,
    color: color,
    height: height,
  );

  if (!kUseBundledKfgqpcFont) return fallback;

  // If the font isn't registered in pubspec.yaml yet, Flutter will silently
  // fall back to the default font.
  return fallback.copyWith(fontFamily: kKfgqpcUthmanTahaNaskhFamily);
}


